node/packages/parser/src/ethereum/index.ts

```diff
@@ -52,21 +57,9 @@ export class TransactionParser {
     const sendAddress = tx.vin[0].addresses?.[0] ?? ''
     const receiveAddress = tx.vout[0].addresses?.[0] ?? ''
 
-    let result: ParseTxSpecific | undefined
-    switch (receiveAddress) {
-      case zrx.PROXY_CONTRACT: {
-        result = this.zrx.parse(tx)
-        break
-      }
-      case this.thor.routerContract: {
-        result = await this.thor.parse(tx)
-        break
-      }
-      case uniV2.ROUTER_CONTRACT: {
-        result = await this.uniV2.parse(tx)
-        break
-      }
-    }
+    const parserResults = await Promise.all(this.parsers.map(async (parser) => await parser.parse(tx)))
```

Comment:
Might be worth using a `for await of` loop so we can just break out of the loop on first successful parse

Reply:
I like where your head is at. I wrote a little find helper that lets us check the result of asynchronous calls, `findAsyncSequential`, that achieves this in a (subjectively) slightly cleaner way - it'll early exit with the first parser to return a result that is not `undefined`.

Commit https://github.com/shapeshift/unchained/pull/238/commits/143f663f3aa13c278ce57bef914db30cc1be0599.

Reply:
It looks like this will run through all parsers, return on the first one that parsed the transaction, and then run await parser?.parse(tx) again. Can we just have the for of loop return the first result we get instead?

That function also seems general enough to have it as a top level parser helper instead of an ethereum helper specifically if we want to keep it as a helper function.

Reply:
Indeed, it was running the `parse()` twice. Luckily idempotent, but a waste of compute.

Fixed in https://github.com/shapeshift/unchained/pull/238/commits/78748721415c9c0242fcb0842bee3b6b7a10b1a7.